app/uk/gov/hmrc/eoricommoncomponent/frontend/forms/AddressDetailsForm.scala

```diff
@@ -40,18 +42,27 @@ object AddressDetailsForm {
     Constraint({
       case s if s.trim.isEmpty => Invalid(ValidationError("cds.subscription.address-details.street.empty.error"))
       case s if s.trim.length > 70 => Invalid(ValidationError("cds.subscription.address-details.street.too-long.error"))
-      case s if !s.matches(noTagsRegex) =>
-        Invalid(ValidationError("cds.subscription.address-details.street.error.invalid-chars"))
+      case s if !s.matches(noTagsRegex) => Invalid(ValidationError("cds.subscription.address-details.street.error.invalid-chars"))
       case _ => Valid
     })
 
   private def validCountry: Constraint[String] =
     Constraint({
       case s if s.trim.isEmpty => Invalid(ValidationError("cds.matching-error.country.invalid"))
-      case s if s.trim.length != Length2 => Invalid(ValidationError("cds.matching-error.country.invalid"))
-      case s if !s.matches(noTagsRegex) =>
-        Invalid(ValidationError("cds.matching.organisation-address.line.error.invalid-chars"))
+      case s if s.trim.length != 2 => Invalid(ValidationError("cds.matching-error.country.invalid"))
+      case s if !s.matches(noTagsRegex) => Invalid(ValidationError("cds.matching.organisation-address.line.error.invalid-chars"))
+      case _ => Valid
+    })
+
+  def validCity: Constraint[String] =
+    Constraint({
+      case s if s.trim.isEmpty => Invalid(ValidationError("cds.subscription.address-details.page-error.city"))
+      case s if s.trim.length > 35 => Invalid(ValidationError("cds.subscription.address-details.page-error.city.too-long"))
+      case s if !s.matches(noTagsRegex) => Invalid(ValidationError("cds.subscription.address-details.city.error.invalid-chars"))
       case _ => Valid
     })
+}
 
+object AddressDetailsForm {
+  val noTagsRegex = "^[^<>]+$"
```

Comment:
just my opinion: would it be better to use only Regex instead of case/if statements? we have a similar address validation in DST as below, maybe we can reuse it?
`val addressRegex = """^[A-Za-z0-9 \-,.&']{1,35}$"""`